queue_services/entity-filer/tests/unit/filing_processors/test_alteration.py

```diff
@@ -199,3 +200,73 @@ async def test_worker_alteration_court_order(app, session, mocker):
+    [
+        (Business.AssociationTypes.CP_COOPERATIVE.value, Business.AssociationTypes.CP_HOUSING_COOPERATIVE.value)
+    ]
+)
+def test_alteration_coop_association_type(app, session, orig_association_type, new_association_type):
+    """Assert that the coop association type is altered."""
+    # setup
+    identifier = 'CP1234567'
+    business = create_business(identifier)
+    business.legal_type = Business.LegalTypes.COOP.value
+
+    alteration_filing = copy.deepcopy(FILING_HEADER)
+    alteration_filing['filing']['business']['legalType'] = Business.LegalTypes.COOP.value
+    alteration_filing['filing']['alteration'] = copy.deepcopy(ALTERATION)
+    alteration_filing['filing']['alteration']['cooperativeAssociationType'] = new_association_type
+    payment_id = str(random.SystemRandom().getrandbits(0x58))
+    filing_submission = create_filing(payment_id, alteration_filing, business_id=business.id)
+
+    filing_meta = FilingMeta()
+
+    # test
+    alteration.process(business=business,
+                       filing_submission=filing_submission,
+                       filing=alteration_filing['filing'],
+                       filing_meta=filing_meta)
+
+    # validate
+    assert business.association_type == new_association_type
+
+
+def test_alteration_coop_rules_and_memorandum(app, session, orig_association_type, new_association_type):
+    """Assert that the coop association type is altered."""
+    # setup
+    identifier = 'CP1234567'
+    business = create_business(identifier)
+    business.legal_type = Business.LegalTypes.COOP.value
+
+    alteration_filing = copy.deepcopy(FILING_HEADER)
+    alteration_filing['filing']['business']['legalType'] = Business.LegalTypes.COOP.value
+    alteration_filing['filing']['alteration'] = copy.deepcopy(ALTERATION)
+    
+    rules_file_key_uploaded_by_user = upload_file('rules.pdf')
+    memorandum_file_key_uploaded_by_user = upload_file('memorandum.pdf')
+    alteration_filing['filing']['incorporationApplication']['cooperative']['rulesFileKey'] = \
+        rules_file_key_uploaded_by_user
+    alteration_filing['filing']['incorporationApplication']['cooperative']['rulesFileName'] = 'Rules_File.pdf'
+    alteration_filing['filing']['incorporationApplication']['cooperative']['memorandumFileKey'] = \
+        memorandum_file_key_uploaded_by_user
+    alteration_filing['filing']['incorporationApplication']['cooperative']['memorandumFileName'] = 'Memorandum_File.pdf'
+    
+    
+    payment_id = str(random.SystemRandom().getrandbits(0x58))
+
+    filing_submission = create_filing(payment_id, alteration_filing, business_id=business.id)
+
+    filing_meta = FilingMeta()
+
+    # test
+    alteration.process(business=business,
+                       filing_submission=filing_submission,
+                       filing=alteration_filing['filing'],
+                       filing_meta=filing_meta)
+
+    business.save()
+
+    # validate
+    assert business.association_type == new_association_type
```

Comment:
can we add assertions to verify the document type, document key and document file_name, filing id are valid for rules and memorandum docs?